File: techila/examples/CSharp/Tutorial/2_parameters/LocalProgram.cs
using System;
using System.Collections.Generic;


namespace LocalProgram
{
    class LocalProgram
    {
        static void Main()
        {
            // Define fixed variables
            int iterations = 5;
            int multip = 2;

            // Define variable to store the result
            int result;

            for (int i = 0; i < iterations; i++)
            {
                // Create new class instance with desired input arguments
                ParametersDist pd = new ParametersDist(multip,i);

                // Call the execute method that performs the multiplication operation
                result = pd.Execute();

                // Print the result of the multiplication on the standard out
                System.Console.WriteLine("Result of iteration {0}: {1}",i + 1, result);
            }


        }



    }
    class ParametersDist // Class containing some executable code
    {
        // Define type of variables that will be used
        int Index;
        int Multip;


        // Define constructor that takes two input arguments
        public ParametersDist(int multip, int index) {
            this.Index=index;
            this.Multip=multip;
        }

        // Define type of 'Result' variable, which will be returned from the 'Execute' method
        int Result;

        // Define an 'Execute' method containing the code that will be executed
        public int Execute()
        {
            // Do simple arithmetic operation
            Result = Multip * Index;

            // Return the result
            return (Result);
        }

    }
}

File: techila/examples/C/Tutorial/2_parameters/local_program.c
// Copyright 2011-2013 Techila Technologies Ltd.

/* This file contains the locally executable program and does not communicate
with the Techila environment in any way. This program will be compiled when
passing the argument 'local_program' to make. */

#include <stdio.h>
#include <fcntl.h>

int main(int argc, char** argv) {
  int loops = 5; // Number of iterations in the for-loop
  int result;
  int i;
  int multip = 2; // Will be used in the multiplication operation in the for-loop
  int myarray[5]; // Array for results

  for (i = 0; i < loops; i++) {
    result = multip * (i + 1); // Perform simple arithmetic operation
    myarray[i] = result; // Store result in array
    printf("Iteration %d, Result: %d\n", i + 1, myarray[i]); // Display progress
  }

  // Display the contents of the array
  printf("Content of the array 'myarray': %i %i %i %i %i\n",
         myarray[0],
         myarray[1],
         myarray[2],
         myarray[3],
         myarray[4]);
  return 0;
}
